src/blocks/remote-data-no-result/render.php

```diff
@@ -0,0 +1,16 @@
+<?php declare(strict_types = 1);
+
+use RemoteDataBlocks\Editor\DataBinding\BlockBindings;
+// Global variables provided by WordPress for block rendering:
+// $attributes (array): The block attributes.
+// $content (string): The block default content.
+// $block (WP_Block): The block instance.
+
+$should_render_empty_result = BlockBindings::should_render_empty_result( $block );
+
+// Skip the rendering if the block's results are not empty.
+if ( ! $should_render_empty_result ) {
+	return;
+}
+
+echo wp_kses_post( $content );
```

Comment:
Only if no results are found would the content of the no results block be rendered.

Reply:
I like all these explanatory comments! Would only say if you think they are useful for PR review then they would also be useful as code comments.

Reply:
I did add a bunch of em already, but I'll go through and add more based on my latest commit